src/util/cilfacade.ml

```diff
@@ -107,12 +163,21 @@ let createCFG (fileAST: file) =
    * See https://github.com/goblint/cil/issues/31#issuecomment-824939793. *)
 
   iterGlobals fileAST (fun glob ->
-      match glob with
-      | GFun(fd,_) ->
-        prepareCFG fd;
-        computeCFGInfo fd true
-      | _ -> ()
-    );
+    match glob with
+    | GFun(fd,_) ->
+      prepareCFG fd;
+    | _ -> ()
+  );
+  (* The unrolling needs to take place between prepareCFG and computeCFGInfo, since prepareCFG handles the breaks     *)
+  (* and continues, which allow us to replicate the loop's body outside of the loop. computeCFG info creates the CFG, *)
+  (* so it needs to happen before that.                                                                               *)
+  if (get_int "exp.unrolling-factor")>0 then loop_unrolling fileAST;
+  iterGlobals fileAST (fun glob ->
+    match glob with
+    | GFun(fd,_) ->
+      computeCFGInfo fd true
+    | _ -> ()
+  );
```

Comment:
I think this could still be done within a single `iterGlobals`, but the loop unrolling visitor should be applied with `visitCilFunction` instead of `visitCilFileSameGlobals`.

Reply:
When I changed the loop unrolling call so it was called after `prepareCFG` it broke the nested loops use case. I was not able to fix it so I changed the old version so it supports `continue` instructions.